src/classFilter.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { resolveClassFilters } from "./languageServerPlugin";
+
+export async function populateStepFilters(config: vscode.DebugConfiguration) {
+    if (!config.stepFilters) {
+        return;
+    }
+
+    const skipClasses = await substituteFilterVariables(config.stepFilters.skipClasses);
+    // Migrate classNameFilters to skipClasses.
+    if (Array.isArray(config.stepFilters.classNameFilters)) {
+        mergeResult(config.stepFilters.classNameFilters, skipClasses);
+    }
+    config.stepFilters.classNameFilters = undefined;
+    config.stepFilters.skipClasses = skipClasses;
+}
+
+export async function substituteFilterVariables(skipClasses: string[]): Promise<any> {
+    if (!skipClasses) {
+        return [];
+    }
+
+    try {
+        // Preprocess skipClasses configurations.
+        if (Array.isArray(skipClasses)) {
+            const index = skipClasses.findIndex((filter) => filter === "$JDK" || filter === "$Libraries");
```

Comment:
How about directly get the boolean value: `const hasReservedName = skipClasses.some((filter) => filter === "$JDK" || filter === "$Libraries");`